javalib/src/main/scala/java/io/File.scala

```diff
@@ -1,9 +1,603 @@
+  private val caseSensitive: Boolean = !Platform.isWindows
+
+  //temporary workaround, while waiting on a working implementation of toLowerCase
+  private def toLowerCaseNaive(str: String): String = {
+    def toLowerChar(c: Char): Char = c match {
+      case 'A' => 'a'
+      case 'B' => 'b'
+      case 'C' => 'c'
+      case 'D' => 'd'
+      case 'E' => 'e'
+      case 'F' => 'f'
+      case 'G' => 'g'
+      case 'H' => 'h'
+      case 'I' => 'i'
+      case 'J' => 'j'
+      case 'K' => 'k'
+      case 'L' => 'l'
+      case 'M' => 'm'
+      case 'N' => 'n'
+      case 'O' => 'o'
+      case 'P' => 'p'
+      case 'Q' => 'q'
+      case 'R' => 'r'
+      case 'S' => 's'
+      case 'T' => 't'
+      case 'U' => 'u'
+      case 'V' => 'v'
+      case 'W' => 'w'
+      case 'X' => 'x'
+      case 'Y' => 'y'
+      case 'Z' => 'z'
+      case x   => x
+    }
+    var chars = str.toCharArray
+    var i     = 0
+    while (i < chars.length) {
+      val lC = toLowerChar(chars(i))
+      chars(i) = lC
+      i += 1
+    }
+    return new String(chars)
+  }
+
+  def listRoots(): Array[File] =
+    if (Platform.isWindows) ???
+    else {
+      var array = new Array[File](1)
+      array(0) = new File("/")
+      return array
+    }
+
+  @throws(classOf[IOException])
+  def createTempFile(prefix: String, suffix: String): File =
+    createTempFile(prefix, suffix, null)
+
+  @throws(classOf[IOException])
+  def createTempFile(prefix: String, suffix: String, directory: File): File =
+    if (prefix == null) throw new NullPointerException
+    else if (prefix.length < 3)
+      throw new IllegalArgumentException("Prefix string too short")
+    else {
+      val tmpDir       = Option(directory).getOrElse(tempDir())
+      val newSuffix    = Option(suffix).getOrElse(".tmp")
+      var result: File = null
+      do {
+        result = genTempFile(prefix, newSuffix, tmpDir)
+      } while (!result.createNewFile())
+      result
+    }
+
+  private def tempDir(): File = {
+    val dir = getenv(c"TMPDIR")
+    if (dir == null) new File("/tmp")
+    else new File(fromCString(dir))
+  }
+
+  private def genTempFile(prefix: String,
+                          suffix: String,
+                          directory: File): File = {
+    val id       = random.nextInt()
+    val fileName = prefix + id + suffix
+    new File(directory, fileName)
+  }
 
-  def compareTo(file: File): scala.Int = ???
 }
+
+// TODO: Extract
+
+//TODO:
+//private def checkURI(uri : URI): Unit
+
+//def deleteOnExit(): Unit = ??? /*atexit{ () => delete() }*/
+
+/*@throws(classOf[IOException])
+private def writeObject(stream: ObjectOutputStream): Unit */
+
+/*@throws(classOf[IOException])
+@throws(classOf[ClassNotFoundException])
+private def readObject(stream: ObjectInputStream): Unit */
+
+//def toURI(): URI
+
+/*@throws(classOf[java.net.MalformedURLException])
+def toURL(): URL*/
+
+/*def File(uri: URI): File = {
+    this()
+    checkURI(uri)
+    path = fixSlashes(uri.getPath())
+}*/
```

Comment:
Please remove all trailing commented-out code apart from the signatures of methods that are not implemented yet. Generally it's also preferable to comment-out methods with `???` on the right hand side like `getChannel`.